### ISSUE TITLE
Display the hash rate

### ISSUE THREAD

**Issue by Susurrus** [Ragnaroek/mithril#7]
It would be most helpful if the hashrate was output at regular intervals. I'm not certain how you ran your hashing tests and got a number, but it'd be useful if one could be output by this tool, even if it was only exposed when setting `RUST_LOG=mithril=info`

**Comment by Ragnaroek** [comment]
Yes, knowing the hash-rate is important. But I have other plans for hash-rate support in mithril.
I'm thinking about an online solution for this.

But not sure yet, please leave this issue open.

**Comment by Susurrus** [comment]
I'm mining currently using `xmr-stak`, and it exposes data over [an HTTP/JSON API](https://github.com/fireice-uk/xmr-stak/blob/master/doc/usage.md#html-and-json-api-report-configuraton). Duplicating this functionality where it makes sense would be useful as there are already many tools that interface with this tool.

**Comment by Titaniumtown** [comment]
Any progress on this? I would love to have a fully functional xmrig alternative in rust!

**Comment by Ragnaroek** [comment]
Not yet, but this is top in my backlog (even before the compiled mode implementation). Otherwise you cannot easily compare the performance with the C implementations (xmrig or xmr-stak). I update the backlog in the Readme to reflect this.

I think going with the approach @Susurrus mentions totally makes sense.

**Comment by RecRanger** [comment]
@Ragnaroek Any thoughts on this? If you merge my other PRs, I'd be happy to take a stab at logging the hashrate to a console at least.

**Comment by Ragnaroek** [comment]
@RecRanger Thanks for your PRs on mithril!
If you want to go ahead and implement console logging this would be a nice addition to mithril and probably a better start than exposing it over http/REST (I never got around to implement that).

Thinking about a cli interface: a (small) TUI interface for mithril would be nice. Maybe based on ratatui or something similiar... This was also something I was thinking about as UI for mithril.
